runtime/databricks/automl_runtime/forecast/utils.py

```diff
@@ -31,17 +32,27 @@ def get_validation_horizon(df: pd.DataFrame, horizon: int, unit: str) -> int:
     :param unit: frequency unit of the time series, which must be a pandas offset alias
     :return: horizon used for validation, in terms of the input `unit`
     """
-    MIN_HORIZONS = 4 # minimum number of horizons in the dataframe
-    df_timedelta = df["ds"].max() - df["ds"].min()
-    horizon_timedelta = pd.to_timedelta(horizon, unit=unit)
+    MIN_HORIZONS = 4 # minimum number of horizons in the datafram
+    horizon_dateoffset = pd.DateOffset(**dict({DATE_OFFSET_KEYWORD_MAP[unit] : horizon}))
+    if unit == 'QS':
+        horizon_dateoffset = horizon_dateoffset*3
 
-    if MIN_HORIZONS * horizon_timedelta <= df_timedelta:
+    if MIN_HORIZONS * horizon_dateoffset + df["ds"].min() <= df["ds"].max():
         return horizon
     else:
-        validation_horizon_timedelta = df_timedelta / MIN_HORIZONS
-        validation_horizon = validation_horizon_timedelta // pd.to_timedelta(1, unit=unit)
-        _logger.info(f"Horizon {horizon_timedelta} too long relative to dataframe's timedelta. Validation horizon will be reduced to {validation_horizon_timedelta}.")
-        return validation_horizon
+        # In order to calculate the validation horizon, we incremental add offset to the start time to
+        # the quater of total timedelta.
+        unit_dateoffset = pd.DateOffset(**dict({DATE_OFFSET_KEYWORD_MAP[unit] : 1}))
+        if unit == 'QS':
+            unit_dateoffset = unit_dateoffset*3
```

Comment:
Will it be better to calculate quarter data with something like
```
DateOffset(months=3)
```
Are these equivalent: `DateOffset(months=3)` and `DateOffset(months=1)*3`?

Reply:
Yes, these are exactly equivalent.